### DISCUSSION TITLE
Displaying a value before it is defined in UI. Updating string when value changes.

### DISCUSSION THREAD

**Discussion by metaperl** [zauberzeug/nicegui#1601] [Q&A]
### Question

Hello, I currently have the following working code:

```py
from nicegui import ui

ui.label('I am')

feeling_toggle = ui.toggle(["feeling", "guessing you are feeling"], value="feeling")

ui.label(f'YOUR NVC SENTENCE: I am {feeling_toggle.value}')

ui.run()
```

However:

1. i would like `ui.label(f'YOUR NVC SENTENCE: I am {feeling_toggle.value}')` to be at the top of the user interface but I get an error that `feeling_toggle` is not defined if i move this line to the top of the file.
2. the output in `ui.label(f'YOUR NVC SENTENCE: I am {feeling_toggle.value}')`  is not updated when `feeling_toggle` is updated. I'm pretty sure the manual will answer one or both of these questions. I will keep digging.

**Comment by natankeddem** [comment]
Here is one way to get it done, I am sure there are other better ways:
```py
from nicegui import ui

label = ui.label('YOUR NVC SENTENCE: I am feeling')

def label_update(value):
    label.text = f'YOUR NVC SENTENCE: I am {value}'
    label.update()

ui.toggle(["feeling", "guessing you are feeling"], value='feeling', on_change=lambda e: label_update(e.value))

ui.run()
```

**Reply by rodja** [reply]
I think your solution is pretty good. You can remove the `label.update()` because it is done automatically by setting the `.text` property.

You could also bind the toggle result to the `ui.label` text:

```py
with ui.row().classes('gap-1'):
    ui.label('YOUR NVC SENTENCE: I am')
    feeling = ui.label()

ui.toggle(["feeling", "guessing you are feeling"], value='feeling').bind_value_to(feeling, 'text')
```
